## Pandas Assignement - Sri Das

In [43]:
import pandas as pd

In [44]:
df = pd.read_csv('https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/ksprojects.csv')

### 1. How many rows of data are in the DataFrame?

In [45]:
rows = df.shape[0]
print('There are {} rows in the DataFrame'.format(rows))

There are 65418 rows in the DataFrame


### 2. What are the names and data types of the columns?

In [46]:
columns = df.columns 
types = df.dtypes
print('Names of the columns are: ', columns)
print ('The date types of the columns are \n', types)

Names of the columns are:  Index(['id', 'name', 'category', 'main_category', 'deadline', 'goal',
       'launched', 'state', 'backers', 'country', 'usd_pledged', 'currency',
       'pledged'],
      dtype='object')
The date types of the columns are 
 id                 int64
name              object
category          object
main_category     object
deadline          object
goal             float64
launched          object
state             object
backers            int64
country           object
usd_pledged      float64
currency          object
pledged          float64
dtype: object


### 3. Do any of the columns contain null values?

In [47]:
df.isnull().sum()

id                 0
name               0
category           0
main_category      0
deadline           0
goal               0
launched           0
state              0
backers            0
country            0
usd_pledged      745
currency           0
pledged            0
dtype: int64

#### The column 'usd_pledged'  containg 745 null values

### 4. Find all successful documentary projects and sort them by the amount pledged. Print the top 10 highest pledges.

In [48]:
df_succ = df[(df['category'] == 'Documentary') & (df['state'] == 'successful')]

In [49]:
df_succ.head()

id                                               name  \
1   1100844465                                       daily digest   
9   1125367305            Meeting Andrei Tarkovsky - Film Website   
24  1216330417  Do It Again: One Man's Quest to Reunite the Kinks   
39  1039240602       The Film Dailies - Live From The Walnut Room   
40  1211984590                               'Mister Rogers & Me'   

       category main_category             deadline     goal  \
1   Documentary  Film & Video  2009-06-01 02:20:00    700.0   
9   Documentary  Film & Video  2009-06-21 02:31:00   1250.0   
24  Documentary  Film & Video  2009-08-10 21:26:00   5500.0   
39  Documentary  Film & Video  2009-09-15 01:00:00    500.0   
40  Documentary  Film & Video  2009-09-20 05:59:00  10000.0   

               launched       state  backers country  usd_pledged currency  \
1   2009-05-04 21:14:28  successful       14      US       700.00      USD   
9   2009-05-30 06:27:42  successful       60      US      2721.00      USD   
24  2009-05-17 05:55:13  successful       79      US      6632.32      USD   
39  2009-08-15 19:52:42  successful        8      US       730.00      USD   
40  2009-08-19 06:16:34  successful      176      US     10795.70      USD   

     pledged  
1     700.00  
9    2721.00  
24   6632.32  
39    730.00  
40  10795.70

In [50]:
df_succ.sort_values('pledged', ascending=False).head(10)

id                                               name  \
11731  1078982332                BRIDEGROOM - An American Love Story   
24166  1162836718  "The Wrecking Crew" The Untold Story of Rock &...   
12127  1326363492                                   The Culture High   
61248  1203470139                         Curious George Documentary   
28003  1205913248                                     ILEANA'S SMILE   
27402  1135999291                             Rewind To Fast-Forward   
16243  1061853312                               Linsanity: the Movie   
27268  1228065368                  El Voto Hispano | The Documentary   
43820  1194449879  PlantPure Nation, a Documentary with Dr. T. Co...   
22237   122290911                          The Killing of Tony Blair   

          category main_category             deadline      goal  \
11731  Documentary  Film & Video  2012-07-20 01:00:28  300000.0   
24166  Documentary  Film & Video  2013-12-21 08:59:00  250000.0   
12127  Documentary  Film & Video  2012-08-04 04:00:00  190000.0   
61248  Documentary  Film & Video  2016-08-25 00:00:00  175000.0   
28003  Documentary  Film & Video  2014-05-21 09:05:00  133000.0   
27402  Documentary  Film & Video  2014-05-01 07:59:00  137000.0   
16243  Documentary  Film & Video  2013-02-16 00:00:00  117000.0   
27268  Documentary  Film & Video  2014-04-27 08:15:23  158000.0   
43820  Documentary  Film & Video  2015-05-14 17:41:41  150000.0   
22237  Documentary  Film & Video  2013-10-07 18:17:14   50000.0   

                  launched       state  backers country  usd_pledged currency  \
11731  2012-06-20 01:00:28  successful     6508      US     384376.0      USD   
24166  2013-10-30 21:06:34  successful     4245      US     313158.0      USD   
12127  2012-06-22 01:40:20  successful     3448      US     240022.0      USD   
61248  2016-07-26 18:01:41  successful     1483      US       2028.0      USD   
28003  2014-04-21 09:05:00  successful     1173      US     177421.0      USD   
27402  2014-03-30 13:35:31  successful     4395      US     176746.0      USD   
16243  2013-01-15 19:46:34  successful     2102      US     167917.0      USD   
27268  2014-03-18 07:15:23  successful      461      US     165401.0      USD   
43820  2015-03-15 16:41:41  successful     1099      US     163930.0      USD   
22237  2013-08-28 18:17:14  successful     4459      GB     255150.0      GBP   

        pledged  
11731  384376.0  
24166  313158.0  
12127  240022.0  
61248  186010.0  
28003  177421.0  
27402  176746.0  
16243  167917.0  
27268  165401.0  
43820  163930.0  
22237  163891.0

### 5. Create a new column named average_per_backer and set its value to the total amount pledged / number of backers.

In [51]:
df['average_per_backer'] = df['pledged']/df['backers']

In [52]:
df.head(20)

id                                               name  \
0   1147015301  "All We Had" Gets Into Cannes -- $10 or More G...   
1   1100844465                                       daily digest   
2   1316334968                             Drive A Faster Car 2.0   
3   1304906577         Accidental to Edinburgh - PHASE 1: AIRFARE   
4   1099226462          Logical Guess Pictures' 2nd Horror Movie!   
5   1301564845  Anatomy of a Credibility Meltdown — a book abo...   
6   1081313636  No-bit: An artist book of analog drawings insp...   
7    112916020          UNINSURED LA™: Support Health Care Reform   
8   1130941392                             Hand made guitar picks   
9   1125367305            Meeting Andrei Tarkovsky - Film Website   
10  1143904247                            It Might Become a Movie   
11  1163942232             Obscurae: a photo show for Braddock PA   
12  1167151653                        Smogr Alert Field Recording   
13  1170035792                One Be Lo - Live in Las Vegas (DVD)   
14  1178729979  Create a Children's Book - to demystify geneti...   
15  1074367099  Help with new things for seeyouinsleep (Canceled)   
16  1291701547                                    American Summer   
17  1001181420                                Sleeping in a Truck   
18  1018181261            The Wild Image Project Shows (Canceled)   
19  1275895636  Create a Children's Book - to demystify diseas...   

            category main_category             deadline     goal  \
0        Documentary  Film & Video  2009-05-20 21:50:00    300.0   
1        Documentary  Film & Video  2009-06-01 02:20:00    700.0   
2         Nonfiction    Publishing  2009-06-04 06:00:00   1000.0   
3            Theater       Theater  2009-06-05 05:59:00   6000.0   
4       Film & Video  Film & Video  2009-06-06 00:45:00    500.0   
5         Nonfiction    Publishing  2009-06-15 02:36:00  30000.0   
6       Illustration           Art  2009-06-15 06:00:00    300.0   
7                Art           Art  2009-06-15 18:00:00   3000.0   
8             Crafts        Crafts  2009-06-16 08:45:00    200.0   
9        Documentary  Film & Video  2009-06-21 02:31:00   1250.0   
10            Shorts  Film & Video  2009-07-01 06:00:00   1000.0   
11       Photography   Photography  2009-07-02 05:59:00    800.0   
12    Graphic Design        Design  2009-07-04 18:40:00    640.0   
13       Documentary  Film & Video  2009-07-05 20:59:00   2500.0   
14  Children's Books    Publishing  2009-07-14 07:01:00   2500.0   
15        Indie Rock         Music  2009-07-14 14:28:00    500.0   
16       Photography   Photography  2009-07-15 04:31:00   1000.0   
17              Rock         Music  2009-07-15 04:58:00  10000.0   
18       Photography   Photography  2009-07-15 05:59:00   5000.0   
19  Children's Books    Publishing  2009-07-15 19:20:00   2500.0   

               launched       state  backers country  usd_pledged currency  \
0   2009-04-30 22:10:30      failed        4      US        40.00      USD   
1   2009-05-04 21:14:28  successful       14      US       700.00      USD   
2   2009-05-04 22:51:31  successful       32      US      1367.00      USD   
3   2009-04-30 22:22:43  successful       24      US      6575.00      USD   
4   2009-04-30 01:32:55  successful       22      US       501.66      USD   
5   2009-05-11 17:34:54      failed        7      US       490.00      USD   
6   2009-05-17 04:53:25      failed        2      US        30.00      USD   
7   2009-05-26 23:40:46      failed        8      US       370.00      USD   
8   2009-05-10 09:02:15      failed        3      US        28.00      USD   
9   2009-05-30 06:27:42  successful       60      US      2721.00      USD   
10  2009-05-29 18:56:20  successful       23      US      1137.00      USD   
11  2009-05-07 21:36:45      failed        2      US        60.00      USD   
12  2009-04-29 22:08:13      failed        3      US        41.00      USD   
13  2009-06-19 08:57:38      failed      

### 6. What happened to the rows with 0 backers? How can this be dealt with?

#### The rows with 0 backers resulted in a null value ('NaN') for average_per_backer. There are two ways of dealing with this, 1. delete the rows with null values or 2. replace the null values with an another number

### 7. Drop all rows with 0 backers then repeat the previous exercise.

In [56]:
df.drop(df[df['backers']==0].index, inplace=True)

In [57]:
df['average_per_backer'] = df['pledged']/df['backers']
df.head(20)

id                                               name  \
0   1147015301  "All We Had" Gets Into Cannes -- $10 or More G...   
1   1100844465                                       daily digest   
2   1316334968                             Drive A Faster Car 2.0   
3   1304906577         Accidental to Edinburgh - PHASE 1: AIRFARE   
4   1099226462          Logical Guess Pictures' 2nd Horror Movie!   
5   1301564845  Anatomy of a Credibility Meltdown — a book abo...   
6   1081313636  No-bit: An artist book of analog drawings insp...   
7    112916020          UNINSURED LA™: Support Health Care Reform   
8   1130941392                             Hand made guitar picks   
9   1125367305            Meeting Andrei Tarkovsky - Film Website   
10  1143904247                            It Might Become a Movie   
11  1163942232             Obscurae: a photo show for Braddock PA   
12  1167151653                        Smogr Alert Field Recording   
13  1170035792                One Be Lo - Live in Las Vegas (DVD)   
14  1178729979  Create a Children's Book - to demystify geneti...   
15  1074367099  Help with new things for seeyouinsleep (Canceled)   
16  1291701547                                    American Summer   
17  1001181420                                Sleeping in a Truck   
20  1024509505                      Produce a "Mean" Feature Film   
21  1317972662         Farmlands an Indie Game Project (Canceled)   

            category main_category             deadline     goal  \
0        Documentary  Film & Video  2009-05-20 21:50:00    300.0   
1        Documentary  Film & Video  2009-06-01 02:20:00    700.0   
2         Nonfiction    Publishing  2009-06-04 06:00:00   1000.0   
3            Theater       Theater  2009-06-05 05:59:00   6000.0   
4       Film & Video  Film & Video  2009-06-06 00:45:00    500.0   
5         Nonfiction    Publishing  2009-06-15 02:36:00  30000.0   
6       Illustration           Art  2009-06-15 06:00:00    300.0   
7                Art           Art  2009-06-15 18:00:00   3000.0   
8             Crafts        Crafts  2009-06-16 08:45:00    200.0   
9        Documentary  Film & Video  2009-06-21 02:31:00   1250.0   
10            Shorts  Film & Video  2009-07-01 06:00:00   1000.0   
11       Photography   Photography  2009-07-02 05:59:00    800.0   
12    Graphic Design        Design  2009-07-04 18:40:00    640.0   
13       Documentary  Film & Video  2009-07-05 20:59:00   2500.0   
14  Children's Books    Publishing  2009-07-14 07:01:00   2500.0   
15        Indie Rock         Music  2009-07-14 14:28:00    500.0   
16       Photography   Photography  2009-07-15 04:31:00   1000.0   
17              Rock         Music  2009-07-15 04:58:00  10000.0   
20    Narrative Film  Film & Video  2009-07-25 02:00:00   6000.0   
21       Video Games         Games  2009-08-01 08:19:00  10000.0   

               launched       state  backers country  usd_pledged currency  \
0   2009-04-30 22:10:30      failed        4      US        40.00      USD   
1   2009-05-04 21:14:28  successful       14      US       700.00      USD   
2   2009-05-04 22:51:31  successful       32      US      1367.00      USD   
3   2009-04-30 22:22:43  successful       24      US      6575.00      USD   
4   2009-04-30 01:32:55  successful       22      US       501.66      USD   
5   2009-05-11 17:34:54      failed        7      US       490.00      USD   
6   2009-05-17 04:53:25      failed        2      US        30.00      USD   
7   2009-05-26 23:40:46      failed        8      US       370.00      USD   
8   2009-05-10 09:02:15      failed        3      US        28.00      USD   
9   2009-05-30 06:27:42  successful       60      US      2721.00      USD   
10  2009-05-29 18:56:20  successful       23      US      1137.00      USD   
11  2009-05-07 21:36:45      failed        2      US        60.00      USD   
12  2009-04-29 22:08:13      failed        3      US        41.00      USD   
13  2009-06-19 08:57:38      failed      

### 8. Create a DataFrame whose index is the first 10 letters of the alphabet and contains two more columns with first 10 prime numbers and the first 10 fibonacci numbers.

In [60]:
ind = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
prime = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
fib = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34]

df1 = pd.DataFrame(list(zip(prime, fib)), index = ind, columns = ['Prime', 'Fibonacci'])

In [61]:
df1.head()

Prime  Fibonacci
a      2          0
b      3          1
c      5          1
d      7          2
e     11          3

### 9. Merge these two DataFrames into a single DataFrame: nycflights13.

In [62]:
flights = pd.read_csv('https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/flights.csv')
airline = pd.read_csv('https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/airlines.csv')

In [68]:
airline.shape

(16, 2)

In [69]:
flights.shape

(336776, 19)

In [71]:
nycflights13 = flights.merge(airline)

In [72]:
nycflights13.head()

year  month  day  dep_time  sched_dep_time  dep_delay  arr_time  \
0  2013      1    1     517.0             515        2.0     830.0   
1  2013      1    1     533.0             529        4.0     850.0   
2  2013      1    1     554.0             558       -4.0     740.0   
3  2013      1    1     558.0             600       -2.0     924.0   
4  2013      1    1     558.0             600       -2.0     923.0   

   sched_arr_time  arr_delay carrier  flight tailnum origin dest  air_time  \
0             819       11.0      UA    1545  N14228    EWR  IAH     227.0   
1             830       20.0      UA    1714  N24211    LGA  IAH     227.0   
2             728       12.0      UA    1696  N39463    EWR  ORD     150.0   
3             917        7.0      UA     194  N29129    JFK  LAX     345.0   
4             937      -14.0      UA    1124  N53441    EWR  SFO     361.0   

   distance  hour  minute            time_hour                   name  
0      1400     5      15  2013-01-01 05:00:00  United Air Lines Inc.  
1      1416     5      29  2013-01-01 05:00:00  United Air Lines Inc.  
2       719     5      58  2013-01-01 05:00:00  United Air Lines Inc.  
3      2475     6       0  2013-01-01 06:00:00  United Air Lines Inc.  
4      2565     6       0  2013-01-01 06:00:00  United Air Lines Inc.

In [73]:
nycflights13.shape

(336776, 20)

### 10. Are there any missing values in this DataFrame? If so, drop them.

In [74]:
nycflights13.isnull().sum()

year                 0
month                0
day                  0
dep_time          8255
sched_dep_time       0
dep_delay         8255
arr_time          8713
sched_arr_time       0
arr_delay         9430
carrier              0
flight               0
tailnum           2512
origin               0
dest                 0
air_time          9430
distance             0
hour                 0
minute               0
time_hour            0
name                 0
dtype: int64

In [77]:
nycflights13_1 = nycflights13.dropna()
nycflights13_1.isnull().sum()

year              0
month             0
day               0
dep_time          0
sched_dep_time    0
dep_delay         0
arr_time          0
sched_arr_time    0
arr_delay         0
carrier           0
flight            0
tailnum           0
origin            0
dest              0
air_time          0
distance          0
hour              0
minute            0
time_hour         0
name              0
dtype: int64

### 11. Merge these two DataFrames into a single DataFrame called census. Please use the following dataset for merging purposes:

In [78]:
df1 = pd.read_csv('https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/state-populations.csv')

df2 = pd.read_csv('https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/census-divisions.csv')

In [84]:
census = df1.merge(df2)
census.head()

state      2010      2011      2012      2013      2014      2015  \
0     Alabama   4785492   4799918   4815960   4829479   4843214   4853875   
1      Alaska    714031    722713    731089    736879    736705    737709   
2     Arizona   6408312   6467163   6549634   6624617   6719993   6817565   
3    Arkansas   2921995   2939493   2950685   2958663   2966912   2977853   
4  California  37332685  37676861  38011074  38335203  38680810  38993940   

       2016 region            division  
0   4863300  South  East South Central  
1    741894   West             Pacific  
2   6931071   West            Mountain  
3   2988248  South  West South Central  
4  39250017   West             Pacific

### 12. Re-shape census so that one column contains all population measures, and another the year attributes.

In [90]:
cen = census.melt(id_vars=['state', 'region', 'division'], var_name='year')
cen.head()

state region            division  year     value
0     Alabama  South  East South Central  2010   4785492
1      Alaska   West             Pacific  2010    714031
2     Arizona   West            Mountain  2010   6408312
3    Arkansas  South  West South Central  2010   2921995
4  California   West             Pacific  2010  37332685

### 13. Group the data by year and summarize it.

In [105]:
cen.groupby('year').describe()

value                                                              \
     count          mean           std       min        25%        50%   
year                                                                     
2010  51.0  6.065651e+06  6.839536e+06  564513.0  1700530.5  4348662.0   
2011  51.0  6.111046e+06  6.907730e+06  567725.0  1713213.0  4369354.0   
2012  51.0  6.156831e+06  6.974942e+06  576765.0  1725818.0  4384799.0   
2013  51.0  6.200096e+06  7.038774e+06  582684.0  1732621.0  4400477.0   
2014  51.0  6.246342e+06  7.110811e+06  583642.0  1741023.0  4413057.0   
2015  51.0  6.292091e+06  7.181603e+06  586555.0  1746940.5  4424611.0   
2016  51.0  6.335834e+06  7.243849e+06  585501.0  1757121.0  4436974.0   

                             
            75%         max  
year                         
2010  6654375.0  37332685.0  
2011  6717221.5  37676861.0  
2012  6776617.0  38011074.0  
2013  6837396.0  38335203.0  
2014  6902053.5  38680810.0  
2015  6988927.5  38993940.0  
2016  7109535.5  39250017.0

### 14. Group the data by region, division and year and summarize it.

In [106]:
cen.groupby(['region', 'division', 'year']).describe()

value                                       \
                                count        mean           std        min   
region  division           year                                              
Midwest East North Central 2010   5.0   9288169.4  3.115322e+06  5690263.0   
                           2011   5.0   9301433.8  3.109807e+06  5709640.0   
                           2012   5.0   9314559.0  3.104953e+06  5726177.0   
                           2013   5.0   9332093.0  3.099620e+06  5742854.0   
                           2014   5.0   9346265.8  3.091076e+06  5758377.0   
...                               ...         ...           ...        ...   
West    Pacific            2012   5.0  10185665.0  1.574263e+07   731089.0   
                           2013   5.0  10274464.0  1.587648e+07   736879.0   
                           2014   5.0  10371286.2  1.601909e+07   736705.0   
                           2015   5.0  10468346.0  1.614496e+07   737709.0   
                           2016   5.0  10560386.6  1.624360e+07   741894.0   

                                                                               
                                       25%        50%         75%         max  
region  division           year                                                
Midwest East North Central 2010  6490528.0  9877495.0  11540983.0  12841578.0  
                           2011  6516480.0  9876213.0  11544824.0  12860012.0  
                           2012  6537743.0  9887238.0  11550839.0  12870798.0  
                           2013  6569102.0  9898982.0  11570022.0  12879505.0  
                           2014  6595233.0  9915767.0  11594408.0  12867544.0  
...                                    ...        ...         ...         ...  
West    Pacific            2012  1391820.0  3899116.0   6895226.0  38011074.0  
                           2013  1406481.0  3925751.0   6968006.0  38335203.0  
                           2014  1416349.0  3968371.0   7054196.0  38680810.0  
                           2015  1425157.0  4024634.0   7160290.0  38993940.0  
                           2016  1428557.0  4093465.0   7288000.0  39250017.0  

[63 rows x 8 columns]

### 15. Create a crosstab between state and category showing amount pledged as the value.

In [107]:
df = pd.read_csv('https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/ksprojects.csv')
df.head()

id                                               name  \
0  1147015301  "All We Had" Gets Into Cannes -- $10 or More G...   
1  1100844465                                       daily digest   
2  1316334968                             Drive A Faster Car 2.0   
3  1304906577         Accidental to Edinburgh - PHASE 1: AIRFARE   
4  1099226462          Logical Guess Pictures' 2nd Horror Movie!   

       category main_category             deadline    goal  \
0   Documentary  Film & Video  2009-05-20 21:50:00   300.0   
1   Documentary  Film & Video  2009-06-01 02:20:00   700.0   
2    Nonfiction    Publishing  2009-06-04 06:00:00  1000.0   
3       Theater       Theater  2009-06-05 05:59:00  6000.0   
4  Film & Video  Film & Video  2009-06-06 00:45:00   500.0   

              launched       state  backers country  usd_pledged currency  \
0  2009-04-30 22:10:30      failed        4      US        40.00      USD   
1  2009-05-04 21:14:28  successful       14      US       700.00      USD   
2  2009-05-04 22:51:31  successful       32      US      1367.00      USD   
3  2009-04-30 22:22:43  successful       24      US      6575.00      USD   
4  2009-04-30 01:32:55  successful       22      US       501.66      USD   

   pledged  
0    40.00  
1   700.00  
2  1367.00  
3  6575.00  
4   501.66

In [124]:
pd.crosstab(index=df['state'], columns=df['category'], values=df['pledged'], aggfunc=sum) 

category    3D Printing   Academic  Accessories    Action   Animals  \
state                                                                 
canceled      106827.50    4364.00    124936.11   13342.0      0.00   
failed        371417.00   81427.74    320990.20   79978.1  15769.87   
live          309215.00      31.00     60904.00       0.0       NaN   
successful   8511414.01  354485.24   2569489.70  226333.9  75497.10   
suspended      10146.00     226.00      3822.00       NaN      0.00   
undefined      13757.00        NaN      2763.00       NaN       NaN   

category     Animation  Anthologies     Apparel       Apps  Architecture  ...  \
state                                                                     ...   
canceled      58910.00      4583.00   228836.00  174745.49      12311.00  ...   
failed       498129.57     20991.00   401102.55  553596.12     139656.99  ...   
live          98403.80      3339.00   236921.59   60101.00       2117.00  ...   
successful  2605326.54   1346909.64  6446055.31  755165.60     460293.08  ...   
suspended         0.00          NaN    56213.00   12989.00        131.00  ...   
undefined          NaN          NaN         NaN        NaN           NaN  ...   

category    Wearables  Weaving        Web   Webcomics   Webseries  \
state                                                               
canceled     239056.0    212.0   88658.00     5668.00    54679.39   
failed       718671.5   6686.0  382408.65    26085.00  1065147.12   
live         141310.0      NaN   57931.00    82120.00      185.00   
successful  7381979.5   6019.0  912191.70  1164701.39  3821840.97   
suspended      9344.0      0.0      50.00         NaN    14878.00   
undefined         NaN      NaN        NaN         NaN         NaN   

category    Woodworking  Workshops  World Music  Young Adult     Zines  
state                                                                   
canceled         153.01        NaN     17136.00       661.00   6940.00  
failed         60328.00     2360.0    120691.83     20920.20  19692.00  
live            1211.73        0.0      9580.00       575.00     80.00  
successful    214568.35    21885.0   1069703.20    145305.62  48668.53  
suspended         91.00        0.0        37.00          NaN       NaN  
undefined           NaN        NaN          NaN          NaN       NaN  

[6 rows x 158 columns]